File: 221028-dailycoding/javaScript/solution.js
export function searchNumber(number, oneCount) {
  const nextNumber = number + 1;

  const nextNumberOneCount = nextNumber.toString(2).match(/1/g).length;
  console.log(`match,  ${nextNumber.toString(2).match(/1/)}`);
  // console.log(`global, ${nextNumber.toString(2).match(/1/g)}`);
  return nextNumberOneCount === oneCount ? nextNumber : searchNumber(number + 1, oneCount);
}
export default function solution(number) {
  const oneCount = number.toString(2).match(/1/g).length;

  return searchNumber(number, oneCount);
}

File: 221028-dailycoding/java/app/src/main/java/Solution.java
class Solution {
  public int solution(int n) {
    int nextNumber = n + 1;

    int oneCount = binaryCount(n);

    while(true) {
      int count = binaryCount(nextNumber);

      if(count == oneCount) {
        break;
      }

      nextNumber += 1;
    }

    return nextNumber;
  }

  public int binaryCount(int n) {
    int count = 0;

    String binaryNumber = Integer.toBinaryString(n);

    for(int i = 0; i<binaryNumber.length(); i += 1) {
      if(binaryNumber.charAt(i) == '1') {
        count += 1;
      }
    }
    return count;
  }
}
